### PR TITLE
Probably fixes medical side effects

### PR DESCRIPTION
**Pull request by quardbreak**
Opening for diff. Ports some medical side effects changes from Infinitystation

### CODE REVIEW

File: code/modules/mob/living/carbon/human/MedicalSideEffects.dm [hunk 1]
```diff
@@ -40,23 +39,21 @@
...
-	var/T = side_effects[name]
-	if (!T)
-		return
-
-	var/datum/medical_effect/M = new T
-	if(M.name == name)
-		M.strength = strength
-		M.start = life_tick
-		side_effects += M
+	var/list/L = subtypesof(/datum/medical_effect)
```
Comment (MistakeNot4892) [inline code/modules/mob/living/carbon/human/MedicalSideEffects.dm]:
```
	for(var/T in subtypesof(/datum/medical_effect))
		var/datum/medical_effect/M = T
		if(initial(M.name) == name)
			M = new T
			M.strength = strength
			M.start = life_tick
			side_effects += M
			break
```
Either this, or you should be deleting `M` if it isn't valid.

File: code/modules/mob/living/carbon/human/MedicalSideEffects.dm [hunk 2]
```diff
@@ -40,23 +39,21 @@
...
 /mob/living/carbon/human/proc/handle_medical_side_effects()
 	//Going to handle those things only every few ticks.
 	if(life_tick % 15 != 0)
-		return 0
+		return FALSE
 
-	var/list/L = typesof(/datum/medical_effect)-/datum/medical_effect
+	var/list/L = subtypesof(/datum/medical_effect)
 	for(var/T in L)
 		var/datum/medical_effect/M = new T
```
Comment (MistakeNot4892) [inline code/modules/mob/living/carbon/human/MedicalSideEffects.dm:58]: As above, these need to use `initial()` or clean up after being instantiated. Preferrably they should be cached somewhere for reference.

File: code/modules/mob/living/carbon/human/MedicalSideEffects.dm [hunk 3]
```diff
@@ -87,15 +84,21 @@
...
 
 /datum/medical_effect/headache/on_life(mob/living/carbon/human/H, strength)
-	var/obj/item/organ/external/head/head = H.get_organ("head")
-	if(istype(head))
+	var/obj/item/organ/external/head/head = H.get_organ(BP_HEAD)
+	if(head)
 		switch(strength)
 			if(1 to 10)
-				H.custom_pain("You feel a light pain in your head.",0, affecting = head)
+				H.custom_pain("You feel a light pain in your head.", 5, affecting = head)
```
Comment (MistakeNot4892) [inline code/modules/mob/living/carbon/human/MedicalSideEffects.dm]: Maybe `in your [head.name]` for these, for species without 'heads'?

File: code/modules/mob/living/carbon/human/MedicalSideEffects.dm [hunk 4]
```diff
@@ -106,13 +109,15 @@
...
-			H.custom_pain("You feel a bit light around the stomach.",0)
-		if(11 to 30)
-			H.custom_pain("Your stomach hurts.",0)
-		if(31 to INFINITY)
-			H.custom_pain("You feel sick.",1)
+	var/obj/item/organ/external/head/groin = H.get_organ(BP_GROIN) //INF
+	if(groin)
+		switch(strength)
+			if(1 to 10)
+				H.custom_pain("You feel a bit light around the stomach.", 10, affecting = groin)
```
Comment (MistakeNot4892) [inline code/modules/mob/living/carbon/human/MedicalSideEffects.dm]: As above, not all species have stomachs.

**Comment by MistakeNot4892** [comment]
Do these actually get used anywhere?

**Comment by comma** [comment]
They're definetly not, not since like 2013 or earlier

File: code/modules/mob/living/carbon/human/MedicalSideEffects.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/modules/mob/living/carbon/human/MedicalSideEffects.dm:58]: If you just add `else qdel(M)` at line 64 this will be ok.

**Review by MistakeNot4892 — CHANGES_REQUESTED** [review]
Just add the qdel and I'll approve this, it's unused code anyway so improvement is harmless.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**